### PR TITLE
Feature/peer 11 appreciation crud

### PR DESCRIPTION
**Pull request by SamnitPatil9882**
Appreciation CRUD
Testing
Swagger

### CODE REVIEW

File: internal/app/appreciation/service.go [hunk 1]
```diff
@@ -0,0 +1,144 @@
...
+
+	//check is corevalue present in database
+	_, err = apprSvc.corevaluesRespo.GetCoreValue(ctx, int64(apprecication.CoreValueID))
+	if err != nil {
+		logger.Error(fmt.Sprintf("err: %v",err))
+		return dto.Appreciation{}, err
+	}
+
+	//check is receiver present in database
+	chk, err := apprSvc.appreciationRepo.IsUserPresent(ctx, tx, apprecication.Receiver)
```
Comment (yama11299) [inline internal/app/appreciation/service.go:91]: can't we check this above and remove it from transaction.

File: internal/app/appreciation/service.go [hunk 2]
```diff
@@ -0,0 +1,144 @@
...
+	infos, pagination, err := apprSvc.appreciationRepo.GetAppreciations(ctx, nil, filter)
+	if err != nil {
+		logger.Error(fmt.Sprintf("err: %v",err))
+		return dto.GetAppreciationResponse{}, err
+	}
+
+	responses := make([]dto.ResponseAppreciation, 0)
+	for _, info := range infos {
+		response := mapRepoGetAppreciationInfoToDTOGetAppreciationInfo(info)
+		responses = append(responses, response)
```
Comment (yama11299) [inline internal/app/appreciation/service.go:136]: responses = append(responses,  mapRepoGetAppreciationInfoToDTOGetAppreciationInfo(info))

File: internal/pkg/dto/appreciation.go [hunk 3]
```diff
@@ -0,0 +1,79 @@
...
+
+import (
+	"strings"
+
+	"github.com/joshsoftware/peerly-backend/internal/pkg/apperrors"
+)
+
+type Appreciation struct {
+	ID                int64  `json:"id"`
+	CoreValueID       int    `json:"core_value_id" `
```
Comment (yama11299) [inline internal/pkg/dto/appreciation.go:11]: again int only?

**Review by yama11299 — CHANGES_REQUESTED** [review]

File: internal/api/appreciation.go [hunk 4]
```diff
@@ -0,0 +1,112 @@
...
+			return
+		}
+
+		res, err := appreciationSvc.DeleteAppreciation(req.Context(), int32(apprId))
+		if err != nil {
+			logger.Error(fmt.Sprintf("err : %v", err))
+			dto.ErrorRepsonse(rw, err)
+			return
+		}
+		if !res {
```
[the remark this answers is not in the document]
Comment (SamnitPatil9882) [reply internal/api/appreciation.go:106]: updated function defination (bool,error) -> error

File: internal/api/helper.go [hunk 5]
```diff
@@ -0,0 +1,61 @@
...
+	if pageStr == "" {
+		page = 1
+	} else {
+		pageInt64, err := strconv.ParseInt(pageStr, 10, 32)
+		if err != nil {
+			logger.Error(fmt.Sprintf("err: %v",err))
+			page = 1 
+		}
+
+		if pageInt64 < 1 {
```
[the remark this answers is not in the document]
Comment (SamnitPatil9882) [reply internal/api/helper.go:25]: In -ve page number page number is default 1
also logic is improved by defining default value

File: internal/pkg/dto/appreciation.go (hunk 3 above)
Comment (SamnitPatil9882) [reply internal/pkg/dto/appreciation.go:11]: udated int to specific

File: internal/api/appreciation_test.go [hunk 6]
```diff
@@ -0,0 +1,310 @@
...
+				mockSvc.On("CreateAppreciation", mock.Anything, dto.Appreciation{
+					ID:1,
+					Description: "Great job!",
+					CoreValueID: 5,
+					TotalRewardPoints: 0,
+					Quarter: 2,
+					Receiver:    2,
+					CreatedAt: 1721631405219,
+					UpdatedAt: 1721631405219,
+				}).Return(dto.Appreciation{}, nil).Once()
```
Comment (yama11299) [inline internal/api/appreciation_test.go:45]: empty response on successful creation?

File: internal/api/helper.go [hunk 7]
```diff
@@ -0,0 +1,67 @@
...
+	// 	if err != nil {
+	// 		logger.Errorf("err: %v",err)
+	// 		page = 1
+	// 	}
+
+	// 	if pageInt64 < 1 {
+	// 		pageInt64 = 1
+	// 	}
+	// 	page = int16(pageInt64)
+	// }
```
Comment (yama11299) [inline internal/api/helper.go:28]: remove commented code

File: internal/api/helper.go [hunk 8]
```diff
@@ -0,0 +1,67 @@
...
+
+	page = 1
+	if pageStr != "" {
+		pageInt64, err := strconv.ParseInt(pageStr, 10, 32)
+		if err != nil {
+			logger.Errorf("err: %v", err)
+		}else if pageInt64 > 0 {
```
Comment (yama11299) [inline internal/api/helper.go:35]: can create separate if instead of else if.

File: internal/api/helper.go [hunk 9]
```diff
@@ -0,0 +1,67 @@
...
+		}
+	}
+
+	limit = 10
+
+	if limitStr != "" {
+		limitInt64, err := strconv.ParseInt(limitStr, 10, 32)
+		if err != nil {
+			logger.Errorf("err: %v", err)
+		}else if limitInt64 > 1000 {
```
Comment (yama11299) [inline internal/api/helper.go:46]: create and use MaxLimit const.

File: internal/api/helper.go [hunk 10]
```diff
@@ -0,0 +1,67 @@
...
+			logger.Errorf("err: %v", err)
+		}else if pageInt64 > 0 {
+			page = int16(pageInt64)
+		}
+	}
+
+	limit = 10
+
+	if limitStr != "" {
+		limitInt64, err := strconv.ParseInt(limitStr, 10, 32)
```
Comment (yama11299) [inline internal/api/helper.go:43]: Don't use Int64 in var name.

File: internal/api/user.go [hunk 11]
```diff
@@ -70,7 +70,7 @@ func listIntranetUsersHandler(userSvc user.Service) http.HandlerFunc {
 
 		page := req.URL.Query().Get("page")
 		if page == "" {
```
Comment (yama11299) [inline internal/api/user.go:72]: Why this is mandatory param? can't we return first page if not mentioned?

File: internal/app/appreciation/helper.go [hunk 12]
```diff
@@ -0,0 +1,67 @@
...
+
+	receiverImageURL := ""
+	if info.ReceiverImageURL.Valid {
+		receiverImageURL = info.ReceiverImageURL.String
+	}
+
+	senderImageURL := ""
+	if info.SenderImageURL.Valid {
+		senderImageURL = info.SenderImageURL.String
+	}
```
Comment (yama11299) [inline internal/app/appreciation/helper.go:33]: can't you remove this?

File: internal/app/appreciation/service.go [hunk 13]
```diff
@@ -0,0 +1,132 @@
...
+type service struct {
+	appreciationRepo repository.AppreciationStorer
+	corevaluesRespo  repository.CoreValueStorer
+}
+
+// Service contains all
+type Service interface {
+	CreateAppreciation(ctx context.Context, appreciation dto.Appreciation) (dto.Appreciation, error)
+	GetAppreciationById(ctx context.Context, appreciationId int32) (dto.ResponseAppreciation, error)
+	ListAppreciations(ctx context.Context, filter dto.AppreciationFilter) (dto.GetAppreciationResponse, error)
```
Comment (yama11299) [inline internal/app/appreciation/service.go:23]: rename to dto.ListAppreciationsResponse

File: internal/app/appreciation/service.go [hunk 14]
```diff
@@ -0,0 +1,132 @@
...
+
+type service struct {
+	appreciationRepo repository.AppreciationStorer
+	corevaluesRespo  repository.CoreValueStorer
+}
+
+// Service contains all
+type Service interface {
+	CreateAppreciation(ctx context.Context, appreciation dto.Appreciation) (dto.Appreciation, error)
+	GetAppreciationById(ctx context.Context, appreciationId int32) (dto.ResponseAppreciation, error)
```
Comment (yama11299) [inline internal/app/appreciation/service.go:22]: rename to dto.GetAppreciationResponse.

File: internal/repository/postgresdb/appreciation.go [hunk 15]
```diff
@@ -0,0 +1,320 @@
...
+	var totalRecords int32
+	err = queryExecutor.QueryRowx(countSql, countArgs...).Scan(&totalRecords)
+	if err != nil {
+		logger.Error("failed to execute count query: ", err.Error())
+		return []repository.AppreciationInfo{}, repository.Pagination{}, apperrors.InternalServerError
+	}
+
+	pagination := getPaginationMetaData(filter.Page, filter.Limit, totalRecords)
+	// Initialize the Squirrel query builder
+	queryBuilder := repository.Sq.Select(
```
Comment (yama11299) [inline internal/repository/postgresdb/appreciation.go:182]: no need to create a new query again, you can use existing queryBuilder and update the Select part only, check this in squirrel documentation once. https://github.com/Masterminds/squirrel/blob/v1.5.4/select.go#L268

File: internal/api/appreciation_test.go (hunk 6 above)
Comment (SamnitPatil9882) [reply internal/api/appreciation_test.go:45]: updated dto.Appreciation{}

File: internal/api/helper.go (hunk 7 above)
Comment (SamnitPatil9882) [reply internal/api/helper.go:28]: removed commented code

File: internal/api/helper.go (hunk 8 above)
Comment (SamnitPatil9882) [reply internal/api/helper.go:35]: no, if we remove if-else and return from if err block we will not reach limit block

File: internal/api/helper.go (hunk 9 above)
Comment (SamnitPatil9882) [reply internal/api/helper.go:46]: created DefaultPageSize in constants
also table names are created in system.go file

File: internal/api/helper.go (hunk 10 above)
Comment (SamnitPatil9882) [reply internal/api/helper.go:43]: updated

File: internal/app/appreciation/helper.go (hunk 12 above)
Comment (SamnitPatil9882) [reply internal/app/appreciation/helper.go:33]: removed

File: internal/app/appreciation/service.go (hunk 14 above)
Comment (SamnitPatil9882) [reply internal/app/appreciation/service.go:22]: updated

File: internal/repository/postgresdb/appreciation.go (hunk 15 above)
Comment (SamnitPatil9882) [reply internal/repository/postgresdb/appreciation.go]: updated

**Comment by yama11299** [comment]
Approved

**Pull request merged**